From 75fe5c08f6adc8274b7814369d52342eea283323 Mon Sep 17 00:00:00 2001
From: Li Jun <b47624@freescale.com>
Date: Thu, 18 Sep 2014 22:59:40 +0800
Subject: [PATCH 0112/5242] MLK-9618-2 usb: chipidea: otg: update is_b_host
 flag when setup&stop host

commit  6ac20b653abb3b95efb6a11eed01a783fb0e6f46 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch sets and clear is_b_host flag of usb_bus when start and stop host in
chipidea host driver.

Acked-by: Peter Chen <peter.chen@freescale.com>
Signed-off-by: Li Jun <b47624@freescale.com>
(cherry picked from commit c164e65922235efcacc3743a9719adc7a9ef7413)
(cherry picked from commit f21e960aba9618c527650cfe3a69cc3ea583ac2a)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/chipidea/host.c |    9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/usb/chipidea/host.c b/drivers/usb/chipidea/host.c
index 876a022..20db44e 100644
--- a/drivers/usb/chipidea/host.c
+++ b/drivers/usb/chipidea/host.c
@@ -291,6 +291,13 @@ static int host_start(struct ci_hdrc *ci)
 		}
 	}
 
+	if (ci_otg_is_fsm_mode(ci)) {
+		if (ci->fsm.id && ci->fsm.otg->state <= OTG_STATE_B_HOST)
+			hcd->self.is_b_host = 1;
+		else
+			hcd->self.is_b_host = 0;
+	}
+
 	ret = usb_add_hcd(hcd, 0, 0);
 	if (ret) {
 		goto disable_reg;
@@ -338,6 +345,8 @@ static void host_stop(struct ci_hdrc *ci)
 		if (ci->platdata->reg_vbus && !ci_otg_is_fsm_mode(ci) &&
 			(ci->platdata->flags & CI_HDRC_TURN_VBUS_EARLY_ON))
 				regulator_disable(ci->platdata->reg_vbus);
+		if (hcd->self.is_b_host)
+			hcd->self.is_b_host = 0;
 	}
 	ci->hcd = NULL;
 	ci->otg.host = NULL;
-- 
1.7.9.5

